### DISCUSSION TITLE
how to update cache after a query

### DISCUSSION THREAD

**Discussion by zhihengGet** [urql-graphql/urql#1784] [Q&A] [answered]
thanks ! 

inside a post query, i also updated /mutated user model so i am doing both Query/mutation in one graphql

 I can split the query and make a mutation for the update , but it means that I need to sent two request instead of just Get Post and update user model together in one graphql query

after the Query , I would like to optimistically update the history cache after issuing a get post query, how can I do it  ? :(

tried resolver but didn't work out,  even though the query response is there, useQuery didn't get it, not sure if cache was mutated too.  
```

resolver {
       Query {
           getPost(){
               cache.updateQuery(...)
               return {__typename:... ,  id:.... }

}
}
}

```
should I split the logic into two graphql request so that I can use updates not resolvers 
 
also is it a good idea to do both mutation/query in one Query ? to save number of query  needed ?

**Comment by kitten** [comment, marked as answer]
it's never a good idea to add side-effects and updates of any kind to queries. They're repeatable and fetch data without changing anything, so they're not meant for updates
